src/Psalm/Internal/Algebra.php

```diff
@@ -33,39 +32,34 @@ class Algebra
      */
     public static function negateTypes(array $all_types): array
     {
-        return array_filter(
-            array_map(
-                /**
-                 * @param  non-empty-list<non-empty-list<Assertion>> $anded_types
-                 *
-                 * @return list<non-empty-list<Assertion>>
-                 */
-                function (array $anded_types): array {
-                    if (count($anded_types) > 1) {
-                        $new_anded_types = [];
-
-                        foreach ($anded_types as $orred_types) {
-                            if (count($orred_types) > 1) {
-                                return [];
-                            }
+        $negated_types = [];
 
-                            $new_anded_types[] = $orred_types[0]->getNegation();
-                        }
+        foreach ($all_types as $key => $anded_types) {
+            if (count($anded_types) > 1) {
+                $new_anded_types = [];
 
-                        return [$new_anded_types];
+                foreach ($anded_types as $orred_types) {
+                    if (count($orred_types) === 1) {
+                        $new_anded_types[] = $orred_types[0]->getNegation();
+                    } else {
+                        continue 2;
                     }
+                }
+
```

Comment:
I tried to debug [errors](https://github.com/vimeo/psalm/runs/5699066638?check_suite_focus=true):

```
ERROR: Trace - src/Psalm/Internal/Algebra.php:48:49 - $anded_types: non-empty-list<non-empty-list<Psalm\Storage\Assertion>> (see https://psalm.dev/224)
                /** @psalm-trace $anded_types */;


ERROR: Trace - src/Psalm/Internal/Algebra.php:49:53 - $new_anded_types: list<Psalm\Storage\Assertion> (see https://psalm.dev/224)
                /** @psalm-trace $new_anded_types */;
```

It looks like `continue 2;` is not properly handled by psalm here.

Reply:
I'd have to take a closer look but I don't have much time these days. Can you add an `assert($new_anded_types !== []);` ? I think this should do the trick :)

Reply:
I'd have to take a closer look but I don't have much time these days. Can you add an `assert($new_anded_types !== []);` ? I think this should do the trick :)

Reply:
Yep, that helps. But that's excessive code (a performance waster) intended just to calm psalm. Tbh, adding this to baseline would be a bit better. I propose to wait a few days other suggestions :)